### ISSUE TITLE
Updates change the order and offset of attributes

### ISSUE THREAD

**Issue by latelatif** [cmu-db/peloton#1223]
While working on delta records for logging, I came across this behavior for updates:
```
postgres=# create table foo(id integer, year integer);
CREATE TABLE
postgres=# insert into foo values(5, 400);
INSERT 0 1
postgres=# select * from foo;
 id | year 
----+------
  5 |  400
(1 row)

postgres=#  update foo set year=year*2, id=id*2;
UPDATE 1
postgres=# select * from foo;
 id  | year 
-----+------
 800 |   10
(1 row)

postgres=#  update foo set year=year*2, id=id*2;
UPDATE 1
postgres=# select * from foo;
 id | year 
----+------
 20 | 1600
(1 row)
```
As can be seen above in the trace, the column values are flipped after the first update and they are flipped again after the second update. It is not just that the values are re-ordered in the output of the select query, the actual offsets in target_list_ are also changed. I just log the column offset and the updated values from codegen for the above operations. The logging information is as follows:

2018-03-23 06:50:29 [src/logging/log_buffer.cpp:88:WriteRecord] INFO  - Offset 1 updated value 800
2018-03-23 06:50:29 [src/logging/log_buffer.cpp:88:WriteRecord] INFO  - Offset 0 updated value 10
2018-03-23 06:50:42 [src/logging/log_buffer.cpp:88:WriteRecord] INFO  - Offset 1 updated value 20
2018-03-23 06:50:42 [src/logging/log_buffer.cpp:88:WriteRecord] INFO  - Offset 0 updated value 1600

**Comment by pervazea** [comment]
It gets more interesting. Sometimes it yields the result above. Sometimes it gets:
```
default_database=# create table tbl (id integer, year integer);
CREATE TABLE
default_database=# INSERT INTO tbl VALUES (5, 400);
INSERT 0 1
default_database=# select * from tbl;
 id | year 
----+------
  5 |  400
(1 row)

default_database=# UPDATE tbl set year=year*2, id=id*2;
UPDATE 2
default_database=# select * from tbl;
 id | year 
----+------
 20 | 1600
(1 row)
```

**Comment by pervazea** [comment]
Which is reported in #1222

**Comment by pervazea** [comment]
This time w/o codegen. It only reports UPDATE 1 this time, but still has the wrong values.
```
default_database=# create table tbl (id integer, year integer);
CREATE TABLE
default_database=# select * from tbl;
 id | year 
----+------
(0 rows)

default_database=# INSERT INTO tbl VALUES (5, 400);
INSERT 0 1
default_database=# select * from tbl;
 id | year 
----+------
  5 |  400
(1 row)

default_database=# UPDATE tbl set year=year*2, id=id*2;
UPDATE 1
default_database=# select * from tbl;
 id | year 
----+------
 20 | 1600
(1 row)
```

**Comment by pervazea** [comment]
I have created a Junit test for this. PR will be coming soon.

**Comment by pervazea** [comment]
This might be an interesting data point. The flipping of the values depends upon ordering in the update SQL statement.

default_database=# create table tbl(id integer, year integer);
CREATE TABLE
default_database=# INSERT INTO tbl VALUES (5, 400);
INSERT 0 1
default_database=# UPDATE tbl set year=year*2, id=id*2;
UPDATE 1
default_database=# select * from tbl;
 id  | year 
-----+------
 800 |   10
(1 row)

default_database=# drop table tbl;
DROP 0
default_database=# create table tbl(id integer, year integer);
CREATE TABLE
default_database=# INSERT INTO tbl VALUES (5, 400);
INSERT 0 1
default_database=# UPDATE tbl set id=id*2, year=year*2;
UPDATE 1
default_database=# select * from tbl;
 id | year 
----+------
 10 |  800
(1 row)

**Comment by poojanilangekar** [comment]
With the Halloween problem fixed,  the bug only exists in the LLVM engine. I think the reason is that it doesn't call the `project_info_->Evaluate` function. A similar logic is used in update_translator.cpp. I think the issue is that the `IsInTarget` function just checks if the `target.first` is equal to the `index`. There is no check for the actual location of the `target_list`, instead it just pushes the value into the vector of values.

**Comment by pervazea** [comment]
Marcel may be able to look at this, so jointly assigning to him (with pmenon to review), unless @pmenon you really want this on your plate.

**Comment by poojanilangekar** [comment]
@tcm-marcel I just fixed the issue. It was a very small fix. If you haven't already started working on it, I can raise a PR with a fix and a test.

**Comment by tcm-marcel** [comment]
@poojanilangekar sure, go ahead - I just wanted to start to look at it

**Comment by pmenon** [comment]
Solved via #1266
